source/development_process/index.rst

```diff
@@ -0,0 +1,306 @@
+breaks something, or if a merge went horribly, horribly wrong.
+
+To set this up, make a ``.travis.yml`` file in the root of the repo. It should
+look something like this::
+
+  language: python
+  python:
+    - "2.7"
+  # command to install dependencies
+  install:
+    - "python setup.py develop"
+    - "pip install flake8"
+  # command to run tests
+  script:
+    - flake8 working_waterfronts/
+    - django-admin test working_waterfronts --settings="working_waterfronts.settings"
+  addons:
+    postgresql: "9.3"
+  before_script:
+    - psql -c 'create database working_waterfronts;' -U postgres
+    - psql -U postgres -c "create extension postgis;" working_waterfronts
+
+Any command put in the ``before_script`` will be run. This allows us to create
+database and extension needed for the app. The ``script`` is the actual test run
+itself. If either command fails, the "build" will fail, and Travis will report
+that it is not safe to merge.
+
+Using flake8 in Travis forces us to keep our code clean and easy to read.
+
+How to Git Out of a Tight Corner
+--------------------------------
+Git is a powerful tool which is generally easy to use, but sometimes it is
+possible to wind up in a weird state. This section of the document describes
+some intermediate level features of git which may be useful to the developer
+team. Developers may also want to read `tricks for configuring git
+<configuring_git.html>`_.
+
+
+- When and when not to force push
+
+Resolving Merge Conflicts
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Hopefully I'll get one in this document so I can add it to this section
+
+Rebasing and Squashing Commits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Often a developer will check out a new branch and while they are working on the
+branch different changes will be merged into develop. To pick up changes on
+develop, check out the working branch and run the following:
+
+.. code::shell
+	$ git checkout fancy-changes
+	$ git rebase develop
+
+This will add all of the changes merged into develop since the branches
+diverged onto ``fancy-new-changes``.
+
+Sometimes a series of commits should be combined into one large commit. This
+can be useful when there were many "work in progress" commits which do not need
+to clutter the git history. This is called squashing commits. First, find the
+oldest commit hash which should be squashed using ``git log``, in this example
+``abcde12345``. Then run:
+
+.. code::shell
+	$ git rebase -i abcde12345 # i stands for interactive
+
+Git will open up the editor and provide detailed instructions on how to choose
+which commits to keep or combine.
+
+Cherry Picking
+~~~~~~~~~~~~~~
+Sometimes it will be necessary to move several commits from one branch to
+another. This can be achieved simply by using ``git cherry-pick``. First, check
+out the branch which the commits will be moved to. Then, find the hash
+representing the commit using ``git log``, for example ``abcde12345``, and run:
```

Comment:
If you don't use `--all` by config, or specify it manually, you can't find the hash after checking out the branch you want to move it to and then running `git log`.